web/src/features/charts/graphUtils.ts

```diff
@@ -236,38 +237,35 @@ const analyzeChartData = (chartData: AreaGraphElement[]) => {
 
 export const getBadgeTextAndIcon = (
   chartData: AreaGraphElement[],
-  t: TFunction
+  t: TFunction,
+  isAggregated: boolean
 ): { text?: string; icon?: LucideIcon } => {
-  const { allTimeSlicerAverageMethod, allEstimated, hasEstimation, estimatedTotal } =
-    analyzeChartData(chartData);
+  const { allTimeSlicerAverageMethod, estimatedTotal } = analyzeChartData(chartData);
+  const estimationTranslation = getEstimationOrAggregationTranslation(
+    t,
+    'pill',
+    isAggregated,
+    undefined,
+    estimatedTotal
+  );
 
   if (estimatedTotal === 0) {
     return {};
   }
 
   if (estimatedTotal) {
     return {
-      text: t('estimation-card.aggregated_estimated.pill', {
-        percentage: estimatedTotal,
-      }),
+      text: estimationTranslation,
       icon: TrendingUpDown,
     };
   }
 
   if (allTimeSlicerAverageMethod) {
     return {
-      text: t(`estimation-card.${EstimationMethods.TSA}.pill`),
+      text: estimationTranslation,
       icon: CircleDashed,
     };
   }
-
-  if (allEstimated) {
-    return { text: t('estimation-badge.fully-estimated'), icon: TrendingUpDown };
-  }
-
-  if (hasEstimation) {
-    return { text: t('estimation-badge.partially-estimated'), icon: TrendingUpDown };
-  }
```

Comment:
These should be handled by the `getEstimationOrAggregationTranslation` logic so the logic can be shared across the app.